tests/test_regression.py

```diff
@@ -17,7 +17,26 @@
     create_rank_mask,
 )
 
-TOLERANCE = 0.01 #max allowed deviation
+
+def assert_dicts_approx_equal(dict1: dict, dict2: dict, tolerance: float = 0.01) -> None:
+    """
+    Compares two dictionaries for approximate equality.
+
+    Args:
+        dict1 (dict): The first dictionary to compare.
+        dict2 (dict): The second dictionary to compare.
+        tolerance (float, optional): The relative tolerance for comparing values. Defaults to 0.01.
+
+    Raises:
+        AssertionError: If the dictionaries are not approximately equal within the specified tolerance.
+    """
+    assert isinstance(dict1, dict) or not isinstance(dict2, dict):
+    
+    assert set(dict1.keys()) == set(dict2.keys()), "The keys in both dictionaries must match."
+    
+    for key in dict1.keys():
+        assert pytest.approx(dict1[key], rel=tolerance) == dict2[key], "Values for key '{key}' are not approximately equal."
```

Comment:
All these separate asserts look a bit like an anti-pattern pytest-wise.
If you do something like `assert dict1 == dict2` , pytest will do all this automatically and give better feedback on a failure.

for example, take this dummy test file:

```
def test_example():
    expected = {"a": 123, "b": "hello world", "c": [1, 2, 3]}
    actual = {"a": 123, "b": "Hello World!", "c": [1, 2, 333], "extra": "meh"}
    assert actual == expected
```

If you run this with pytest, you get a report like this:
```
    def test_example():
        expected = {"a": 123, "b": "hello world", "c": [1, 2, 3]}
        actual = {"a": 123, "b": "Hello World!", "c": [1, 2, 333], "extra": "meh"}
>       assert actual == expected
E       AssertionError: assert {'a': 123, 'b...extra': 'meh'} == {'a': 123, 'b...c': [1, 2, 3]}
E
E         Omitting 1 identical items, use -vv to show
E         Differing items:
E         {'c': [1, 2, 333]} != {'c': [1, 2, 3]}
E         {'b': 'Hello World!'} != {'b': 'hello world'}
E         Left contains 1 more item:
E         {'extra': 'meh'}
E         Use -v to get more diff
```

Notice how pytest does not just say that the dicts are different, but concretely points out the differing details: which keys are differing, what values are differing, etc.

If you even add the `-vv` option to `pytest` command, pytest will even do more effort and analyze nested constructs to more detail:
```
E       AssertionError: assert {'a': 123, 'b': 'Hello World!', 'c': [1, 2, 333], 'extra': 'meh'} == {'a': 123, 'b': 'hello world', 'c': [1, 2, 3]}
E
E         Common items:
E         {'a': 123}
E         Differing items:
E         {'c': [1, 2, 333]} != {'c': [1, 2, 3]}
E         {'b': 'Hello World!'} != {'b': 'hello world'}
E         Left contains 1 more item:
E         {'extra': 'meh'}
E
E         Full diff:
E           {
E               'a': 123,
E         -     'b': 'hello world',
E         ?           ^     ^
E         +     'b': 'Hello World!',
E         ?           ^     ^    +
E               'c': [
E                   1,
E                   2,
E         -         3,
E         +         333,
E         ?         ++
E               ],
E         +     'extra': 'meh',
E           }
```

Reply:
what you could however is make a helper to construct the dictionary with `pytest.approx`ed expectation, e.g. see https://github.com/Open-EO/openeo-python-driver/blob/c0110d70185687aae5b9144d87f36a72c914949f/openeo_driver/testing.py#L500-L519